### ISSUE TITLE
Save statistics of last command to JSON file

### ISSUE THREAD

**Issue by robaca** [vercel/turborepo#1424]
### Describe the feature you'd like to request

We are currently setting up a monorepo and - as a last step - want to manually decide about how to continue (release and deploy, or abort). For this we need more details about the last execution results. 
- Did everything work, only the final webdriver tests of one or more packages fail?
- Did it fail because of flakyness or because of a real error (currently needs a human to check it)

I would like to be able to analyze last executions of a turbo command 
- what steps of which packages were executed
- which had cache hits
- which steps succeeded, failed or were skipped (in combination with the `--continue` option)
- how long did each of the steps take

### Describe the solution you'd like

Always or only via an additional option, enable writing the above statistics into a JSON file, which we can analyze via `jq` or any other tool.

This info could also be used for a second feature - providing a human readable summary.

### Describe alternatives you've considered

* Parsing human readable turbo console output to extract the necessary infos - doesn't seem like a viable option.
* Running only the "deploy-and-test" steps of all packages in a separate `turbo` command would help, but also would mean having to wait for all previous steps of all packages to finish before starting the tests. We would prefer to start the webdriver tests on a per project basis as early as possible.

**Comment by robaca** [comment]
Another scenario:
* Detect which deployment steps were run and successful, so that at a later time, an undeploy for the same packages can be triggered.

**Comment by gsoltis** [comment]
I think this is a good idea, I we're going to track it with #1368

**Comment by gsoltis** [comment]
Marking as duplicate of #1368
